pkg/apis/options/legacy_options.go

```diff
@@ -93,6 +97,14 @@ func (l *LegacyOptions) ToOptions() (*Options, error) {
 	}
 	l.Options.Providers = providers
 
+	// in case of legacy options, we initilize single tenant implementation of tenant loader
+	l.Options.TenantLoader = tenant.LoaderConfiguration{
+		Type: "single",
+		Single: &single.Configuration{
+			Provider: &providers[0],
+		},
+	}
```

Comment:
This kind of initialisation shouldn't be happening within the options part of the project

Reply:
agreed,

Reply:
resolved, removed the initialization logic away from the `options` package.